### PR TITLE
docs: add agent setup

### PR DESCRIPTION
**Pull request by MosCD3**
As usual, needs some linguistics review

### CODE REVIEW

File: docs/getting-started/0-agent.md [hunk 1]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+  }
+}
+
+//Somewhere in the code (before init Agent)
+const genesis = await downloadGenesis(GENESIS_URL);
+const genesisPath = await storeGenesis(genesisString, 'genesis.txn');
+
+```
+
+Note: You do not need genesis file if you are creating a connection between your agent and other agent for exchanging simple messages.
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:114]:
```suggestion
Note: You do not need the genesis file if you are creating a connection between your Agent and another Agent for exchanging simple messages.
```

File: docs/getting-started/0-agent.md [hunk 2]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+
+### 3- Get Mediator Connection URL (Optional)
+
+Mediators (Routing Agents) are Agents that serve as intermediaries to facilitate the flow of messages between other types of agents.
+You will need mediator agent if you are going to deal with VC (Verifiable Credentials), however, you can ignore the mediator step if you are creating an Agent for merely connecting directly to other Agent for exchanging messages.
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:119]:
```suggestion
You will need a mediator Agent if you are going to deal with VC (Verifiable Credentials), however, you can ignore the mediator step if you are creating an Agent for the sole purpose of exchanging messages with another Agent. 
```

File: docs/getting-started/0-agent.md [hunk 3]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+
+Example: If you are testing with VCs, you can use [Indicio Public Mediator](https://indicio-tech.github.io/mediator/)
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:121]:
```suggestion
Example: If you are testing VC related functionality and need a mediator, you can use the [Indicio Public Mediator](https://indicio-tech.github.io/mediator/).
```

File: docs/getting-started/0-agent.md [hunk 4]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+- Head to [Indicio Public Mediator](https://indicio-tech.github.io/mediator/)
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:122]:
```suggestion
- Head to [Indicio Public Mediator](https://indicio-tech.github.io/mediator/).
```

File: docs/getting-started/0-agent.md [hunk 5]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+- Copy mediator invite url and save it (i.e. MEDIATOR_INVITE = "url")
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:123]:
```suggestion
- Copy mediator invite url and save it (i.e. MEDIATOR_INVITE = "url").
```

File: docs/getting-started/0-agent.md [hunk 6]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+
+Note: if the invite url use insecure http you will have to adjust your code to allow insecure traffic
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:125]:
```suggestion
Note: If the invite url uses insecure http you will have to adjust your code to allow for insecure traffic.
```

File: docs/getting-started/0-agent.md [hunk 7]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+- Instructions for [iOS](https://stackoverflow.com/questions/31254725/transport-security-has-blocked-a-cleartext-http)
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:126]:
```suggestion
- Instructions for [iOS](https://stackoverflow.com/questions/31254725/transport-security-has-blocked-a-cleartext-http).
```

File: docs/getting-started/0-agent.md [hunk 8]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+- Instructions for [Android](https://stackoverflow.com/questions/51902629/how-to-allow-all-network-connection-types-http-and-https-in-android-9-pie)
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:127]:
```suggestion
- Instructions for [Android](https://stackoverflow.com/questions/51902629/how-to-allow-all-network-connection-types-http-and-https-in-android-9-pie).
```

File: docs/getting-started/0-agent.md [hunk 9]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+
+More about [Mediators](3-routing.md)
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:129]:
```suggestion
More about [Mediators](3-routing.md).
```

File: docs/getting-started/0-agent.md [hunk 10]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+### 5- Create Transports
+
+After creating the Agent object, we need to create and set the Inbound/Outbound transports that will handle traffic to/from the Agent.
+
+
+```ts
+  const httpOutboundTransporter = new HttpOutboundTransporter();
+  agent.setOutboundTransporter(httpOutboundTransporter);
+
+  //Inbound transport are currently built-in and you don't need to create it
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:158]:
```suggestion
  //Inbound transports are currently built-in, you don't need to create them.
```

File: docs/getting-started/0-agent.md [hunk 11]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+```
+
+More about [Transports](1-transports.md)
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:161]:
```suggestion
More about [Transports](1-transports.md).
```

File: docs/getting-started/0-agent.md [hunk 12]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+
+
+### 6- Init Agent
+
+After creating the Agent object and configuring it with all configuration, we initialize the Agent
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:166]:
```suggestion
After creating the Agent object and configuring it, we initialize the Agent.
```

File: docs/getting-started/0-agent.md [hunk 13]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+  console.log('Initialized agent!');
+}catch(e){
+  console.log(`Agent init error:${e}`);
+}
+```
+
+
+### 6- Handling State Changes
+
+After you successfully initialize your Agent, you will notice most of the hard work is done by the underlying Aries/Indy framework however, you as a controller need to intercept at some situations lets say when you need to accept an offered credential or reject/accept an incoming connection request. This is done through state change handlers
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:183]:
```suggestion
After you successfully initialize your Agent, you will notice that most of the hard work is being done by the underlying Aries/Indy framework. However, you as a controller will need to intercept in some situations when there is a state change (like when you need to accept an offered credential or reject/accept an incoming connection request). This is done through state change handlers.
```

File: docs/getting-started/0-agent.md [hunk 14]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+
+agent.events.on<ConnectionStateChangedEvent>(
+  ConnectionEventTypes.ConnectionStateChanged,
+  event => {
+    handleConnectionStateChange(event); //custom callback for handling any state change when offering/receiving a connection
+  },
+);
+```
+
+Example: This sample Credential callback shows how to detect that a credential is received and show the user the credential asserts and give the option to accept/reject the offered credential
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:209]:
```suggestion
Example: This sample credential callback shows how to detect that a credential is received, show the user the credential asserts and give the option to accept/reject the offered credential.
```

File: docs/getting-started/0-agent.md [hunk 15]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+
+    //Credential recieved and stored in the wallet
+    console.log('>> Received Credential');
+    await agent.credentials.acceptCredential(event.payload.credentialRecord.id);
+    console.log('ALL DONE - CREDENTAIL ACCEPTED');
+  }
+};
+```
+
+More about [Credentials](5-credentials.md)
```
Comment (anagoessens) [inline docs/getting-started/0-agent.md:238]:
```suggestion
More about [Credentials](5-credentials.md).
```

**Review by anagoessens — COMMENTED** [review]
Some language/grammar suggestions.

File: docs/getting-started/0-agent.md [hunk 16]
```diff
@@ -27,6 +27,218 @@ const agentConfig: InitConfig = {
...
+} from '@aries-framework/core'
+import { agentDependencies } from '@aries-framework/node'
+import RNFS from 'react-native-fs'
+```
+
+### 2- Download genesis file (Optional)
+
+In working with an Indy network, the ledger’s genesis file contains the information necessary for an agent to connect to that ledger, in other words the genesis file contains the schema (like the DB schema) of the target indy node ledger you are attempting to connect to. 
+When will you need genesis file?, you will need the genesis file of the indy ledger you will connect to, to accept/verify credentials.
+Example: Lets say your agent will accept a verifiable credential from trinsic.id, you will probably need to download the genesis file for Sovrin network
```
[the remark this answers is not in the document]
Comment (blu3beri) [reply docs/getting-started/0-agent.md:59]: let's

File: docs/getting-started/0-agent.md [hunk 17]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+    console.log('EXCEPTION> storeGenesis:', e);
+    return '';
+  }
+}
+
+//Somewhere in the code (before init Agent)
+const genesis = await downloadGenesis(GENESIS_URL);
+const genesisPath = await storeGenesis(genesisString, 'genesis.txn');
+
+```
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:112]: I think we shouldn't encourage people to download a genesis transaction file on each startup. I'd recommend the following approach: 

This also removes the need to use platform specific file system libraries (as it is handled by the framework)

```ts
const BCOVRIN_TEST_GENESIS = `{"reqSignature":{},"txn":{"data":{"data":{"alias":"Node1","blskey":"4N8aUNHSgjQVgkpm8nhNEfDf6txHznoYREg9kirmJrkivgL4oSEimFF6nsQ6M41QvhM2Z33nves5vfSn9n1UwNFJBYtWVnHYMATn76vLuL3zU88KyeAYcHfsih3He6UHcXDxcaecHVz6jhCYz1P2UZn2bDVruL5wXpehgBfBaLKm3Ba","blskey_pop":"RahHYiCvoNCtPTrVtP7nMC5eTYrsUA8WjXbdhNc8debh1agE9bGiJxWBXYNFbnJXoXhWFMvyqhqhRoq737YQemH5ik9oL7R4NTTCz2LEZhkgLJzB3QRQqJyBNyv7acbdHrAT8nQ9UkLbaVL9NBpnWXBTw4LEMePaSHEw66RzPNdAX1","client_ip":"138.197.138.255","client_port":9702,"node_ip":"138.197.138.255","node_port":9701,"services":["VALIDATOR"]},"dest":"Gw6pDLhcBcoQesN72qfotTgFa7cbuqZpkX3Xo6pLhPhv"},"metadata":{"from":"Th7MpTaRZVRYnPiabds81Y"},"type":"0"},"txnMetadata":{"seqNo":1,"txnId":"fea82e10e894419fe2bea7d96296a6d46f50f93f9eeda954ec461b2ed2950b62"},"ver":"1"}
{"reqSignature":{},"txn":{"data":{"data":{"alias":"Node2","blskey":"37rAPpXVoxzKhz7d9gkUe52XuXryuLXoM6P6LbWDB7LSbG62Lsb33sfG7zqS8TK1MXwuCHj1FKNzVpsnafmqLG1vXN88rt38mNFs9TENzm4QHdBzsvCuoBnPH7rpYYDo9DZNJePaDvRvqJKByCabubJz3XXKbEeshzpz4Ma5QYpJqjk","blskey_pop":"Qr658mWZ2YC8JXGXwMDQTzuZCWF7NK9EwxphGmcBvCh6ybUuLxbG65nsX4JvD4SPNtkJ2w9ug1yLTj6fgmuDg41TgECXjLCij3RMsV8CwewBVgVN67wsA45DFWvqvLtu4rjNnE9JbdFTc1Z4WCPA3Xan44K1HoHAq9EVeaRYs8zoF5","client_ip":"138.197.138.255","client_port":9704,"node_ip":"138.197.138.255","node_port":9703,"services":["VALIDATOR"]},"dest":"8ECVSk179mjsjKRLWiQtssMLgp6EPhWXtaYyStWPSGAb"},"metadata":{"from":"EbP4aYNeTHL6q385GuVpRV"},"type":"0"},"txnMetadata":{"seqNo":2,"txnId":"1ac8aece2a18ced660fef8694b61aac3af08ba875ce3026a160acbc3a3af35fc"},"ver":"1"}
{"reqSignature":{},"txn":{"data":{"data":{"alias":"Node3","blskey":"3WFpdbg7C5cnLYZwFZevJqhubkFALBfCBBok15GdrKMUhUjGsk3jV6QKj6MZgEubF7oqCafxNdkm7eswgA4sdKTRc82tLGzZBd6vNqU8dupzup6uYUf32KTHTPQbuUM8Yk4QFXjEf2Usu2TJcNkdgpyeUSX42u5LqdDDpNSWUK5deC5","blskey_pop":"QwDeb2CkNSx6r8QC8vGQK3GRv7Yndn84TGNijX8YXHPiagXajyfTjoR87rXUu4G4QLk2cF8NNyqWiYMus1623dELWwx57rLCFqGh7N4ZRbGDRP4fnVcaKg1BcUxQ866Ven4gw8y4N56S5HzxXNBZtLYmhGHvDtk6PFkFwCvxYrNYjh","client_ip":"138.197.138.255","client_port":9706,"node_ip":"138.197.138.255","node_port":9705,"services":["VALIDATOR"]},"dest":"DKVxG2fXXTU8yT5N7hGEbXB3dfdAnYv1JczDUHpmDxya"},"metadata":{"from":"4cU41vWW82ArfxJxHkzXPG"},"type":"0"},"txnMetadata":{"seqNo":3,"txnId":"7e9f355dffa78ed24668f0e0e369fd8c224076571c51e2ea8be5f26479edebe4"},"ver":"1"}
{"reqSignature":{},"txn":{"data":{"data":{"alias":"Node4","blskey":"2zN3bHM1m4rLz54MJHYSwvqzPchYp8jkHswveCLAEJVcX6Mm1wHQD1SkPYMzUDTZvWvhuE6VNAkK3KxVeEmsanSmvjVkReDeBEMxeDaayjcZjFGPydyey1qxBHmTvAnBKoPydvuTAqx5f7YNNRAdeLmUi99gERUU7TD8KfAa6MpQ9bw","blskey_pop":"RPLagxaR5xdimFzwmzYnz4ZhWtYQEj8iR5ZU53T2gitPCyCHQneUn2Huc4oeLd2B2HzkGnjAff4hWTJT6C7qHYB1Mv2wU5iHHGFWkhnTX9WsEAbunJCV2qcaXScKj4tTfvdDKfLiVuU2av6hbsMztirRze7LvYBkRHV3tGwyCptsrP","client_ip":"138.197.138.255","client_port":9708,"node_ip":"138.197.138.255","node_port":9707,"services":["VALIDATOR"]},"dest":"4PS3EDQ3dW1tci1Bp6543CfuuebjFrg36kLAUcskGfaA"},"metadata":{"from":"TWwCRQRZ2ZHMJFn9TzLp7W"},"type":"0"},"txnMetadata":{"seqNo":4,"txnId":"aa5e817d7cc626170eca175822029339a444eb0ee8f0bd20d3b0b76e566fb008"},"ver":"1"}`

const agentConfig = {
   poolName: 'BCovrin Test'
   genesisTransactions: BCOVRIN_TEST_GENESIS
}
```

File: docs/getting-started/0-agent.md [hunk 18]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+
+Note: You do not need the genesis file if you are creating a connection between your Agent and another Agent for exchanging simple messages.
+
+### 3- Get Mediator Connection URL (Optional)
+
+Mediators (Routing Agents) are Agents that serve as intermediaries to facilitate the flow of messages between other types of agents.
+You will need a mediator Agent if you are going to deal with VC (Verifiable Credentials), however, you can ignore the mediator step if you are creating an Agent for the sole purpose of exchanging messages with another Agent.
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:119]: This assumes mobile environments. Mediators are only needed when you can't host an endpoint to receive messages yourself (this is the case on mobile environments). It is required for all exchanges on mobile, not only when dealing with VCs

File: docs/getting-started/0-agent.md [hunk 19]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+Mediators (Routing Agents) are Agents that serve as intermediaries to facilitate the flow of messages between other types of agents.
+You will need a mediator Agent if you are going to deal with VC (Verifiable Credentials), however, you can ignore the mediator step if you are creating an Agent for the sole purpose of exchanging messages with another Agent. 
+
+Example: If you are testing VC related functionality and need a mediator, you can use the [Indicio Public Mediator](https://indicio-tech.github.io/mediator/).
+- Head to [Indicio Public Mediator](https://indicio-tech.github.io/mediator/).
+- Copy mediator invite url and save it (i.e. MEDIATOR_INVITE = "url").
+
+Note: If the invite url uses insecure http you will have to adjust your code to allow for insecure traffic.
+- Instructions for [iOS](https://stackoverflow.com/questions/31254725/transport-security-has-blocked-a-cleartext-http).
+- Instructions for [Android](https://stackoverflow.com/questions/51902629/how-to-allow-all-network-connection-types-http-and-https-in-android-9-pie).
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:127]: This also assumes mobile environments. I'm okay with targeting mobile environments but we should make it a getting started in React Native guide in that case

File: docs/getting-started/0-agent.md [hunk 20]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+
+More about [Mediators](3-routing.md).
+
+### 4- Create Agent
+
+```ts
+const agentConfig: InitConfig = {
+      label: 'my-agent9',
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:135]:
```suggestion
      label: 'My Agent',
```

File: docs/getting-started/0-agent.md [hunk 21]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+      mediatorConnectionsInvite: MEDIATOR_INVITE,
+      walletConfig: {id: 'walletId10'},
+      walletCredentials: {key: 'testkey0230482304823042304244'},
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:138]:
```suggestion
      walletConfig: {
      	id: 'WalletId',
      	key: 'TestKey',
      },
```

File: docs/getting-started/0-agent.md [hunk 22]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+      autoAcceptConnections: true,
+      poolName: 'test-193',
+      genesisPath,
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:141]:
```suggestion
      poolName: 'BCovrin Test',
      genesisTransactions: BCOVRIN_TEST_GENESIS,
```

File: docs/getting-started/0-agent.md [hunk 23]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+
+After creating the Agent object, we need to create and set the Inbound/Outbound transports that will handle traffic to/from the Agent.
+
+
+```ts
+  const httpOutboundTransporter = new HttpOutboundTransporter();
+  agent.setOutboundTransporter(httpOutboundTransporter);
+
+  //Inbound transports are currently built-in, you don't need to create them.
+```
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:159]:
```suggestion

### Outbound transports

Outbound transport are for sending messages. Outbound transports can optionally be used for receiving messages (e.g. in the case of a mediator). Currently two transports are supported as part of the framework, HTTP and WebSockets.

\`\`\`ts
  // For sending messages to http endpoints
  const httpOutboundTransport = new HttpOutboundTransport();
  agent.registerOutboundTransport(httpOutboundTransport);
  
  // For sending messages to ws endpoints
  const wsOutboundTransport = new WsOutboundTransport();
  agent.registerOutboundTransport(wsOutboundTransport);
\`\`\`

### Inbound transports

Inbound transports are for receiving messages. The inbound transports are only available in NodeJS environments, as we need to start a server to expose an HTTP endpoint and WebSocket server. You can import the inbound transports from the `@aries-framework/node` package

\`\`\`ts
  // For receiving messages on an HTTP endpoint
  const httpInboundTransport = new HttpInboundTransport({ port: 3000 });
  agent.registerInboundTransport(httpInboundTransport);
  
  // For receiving messages on an WS endpoint
  const wsInboundTransport = new WsInboundTransport({ port: 30001 });
  agent.registerInboundTransport(wsInboundTransport);
\`\`\`
```

Comment (TimoGlastra) [reply docs/getting-started/0-agent.md:159]: Please remove the backticks, but github was struggling with the backtikcs in the suggestions

File: docs/getting-started/0-agent.md [hunk 24]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+More about [Transports](1-transports.md).
+
+
+### 6- Init Agent
+
+After creating the Agent object and configuring it, we initialize the Agent.
+
+```ts
+
+//Its highly recommended to wrap the initialization flow in a try/catch block
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:170]:
```suggestion
// It's highly recommended to wrap the initialization flow in a try/catch block 
```

File: docs/getting-started/0-agent.md [hunk 25]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+  await agent.initialize();
+  console.log('Initialized agent!');
+}catch(e){
+  console.log(`Agent init error:${e}`);
 }
+```
+
 
-const agent = new Agent(agentConfig)
+### 6- Handling State Changes
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:181]:
```suggestion
### 7- Handling State Changes
```

File: docs/getting-started/0-agent.md [hunk 26]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+}
+
+const agent = new Agent(agentConfig, agentDependencies)
+```
+## Complete Agent Initialization
+
+This is the optimal initialization code for a scenario where complete functionality is needed.
+We will walk through the following steps to initialize the agent with full capabilities.
+
+### 1- Import statements
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:35]: I'd prefer to use a dot, or add a space between the number or hyphen

```suggestion
### 1. Import statements
```

```suggestion
### 1 - Import statements
```

File: docs/getting-started/0-agent.md [hunk 27]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+    handleCredentialStateChange(event); //custom callback for handling any state change when offering/receiving VCs
+  },
+);
+
+agent.events.on<ConnectionStateChangedEvent>(
+  ConnectionEventTypes.ConnectionStateChanged,
+  event => {
+    handleConnectionStateChange(event); //custom callback for handling any state change when offering/receiving a connection
+  },
+);
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:206]:
```suggestion
agent.events.on<BasicMessageReceivedEvent>(
	AgentEventTypes.AgentMessageReceived,
    handleBasicMessageReceive
)

agent.events.on<CredentialStateChangedEvent>(
  CredentialEventTypes.CredentialStateChanged,
  // Custom callback for handling any state change when offering/receiving VCs
  handleCredentialStateChange
)

agent.events.on<ConnectionStateChangedEvent>(
  ConnectionEventTypes.ConnectionStateChanged,
  // Custom callback for handling any state change when offering/receiving a connection
  handleConnectionStateChange(event)
)
```

File: docs/getting-started/0-agent.md [hunk 28]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+
+```ts
+const handleCredentialStateChange = async (
+  event: CredentialStateChangedEvent,
+) => {
+  console.log(
+    `>> Credential state changed: ${event.payload.credentialRecord.id}, previous state -> ${event.payload.previousState} new state: ${event.payload.credentialRecord.state}`,
+  );
+
+  if (event.payload.credentialRecord.state === 'offer-received') {
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:219]:
```suggestion
  if (event.payload.credentialRecord.state === CredentialState.OfferReceived) {
```

File: docs/getting-started/0-agent.md [hunk 29]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+    console.log(`>> Received offer, should display credential to user`);
+    
+    //Display offer to user 
+    //..
+    //On user click "Accept"
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:224]:
```suggestion
    // Display offer to user 
    // On user click "Accept"
```

File: docs/getting-started/0-agent.md [hunk 30]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+
+### 4- Create Agent
+
+```ts
+const agentConfig: InitConfig = {
+      label: 'my-agent9',
+      mediatorConnectionsInvite: MEDIATOR_INVITE,
+      walletConfig: {id: 'walletId10'},
+      walletCredentials: {key: 'testkey0230482304823042304244'},
+      autoAcceptConnections: true,
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:139]: We now have an auto accept "ContentApproved" that I think is good enough for the getting started guide. We could then explain how to manually handle the flow in the in-depth guides

```suggestion
      autoAcceptConnections: true,
      autoAcceptCredentials: AutoAcceptCredential.ContentApproved,
      autoAcceptProofs: AutoAcceptProof.ContentApproved,
```

File: docs/getting-started/0-agent.md [hunk 31]
```diff
@@ -13,37 +13,241 @@ Before initializing your agent, make sure you have followed the setup guide for
...
+    console.log(`>>ACCEPTING OFFER`);
+    agent.credentials.acceptOffer(event.payload.credentialRecord.id);
+
+  } else if (event.payload.credentialRecord.state === 'credential-received') {
+
+    //Credential recieved and stored in the wallet
+    console.log('>> Received Credential');
+    await agent.credentials.acceptCredential(event.payload.credentialRecord.id);
+    console.log('ALL DONE - CREDENTAIL ACCEPTED');
+  }
```
Comment (TimoGlastra) [inline docs/getting-started/0-agent.md:234]: With content approved mode we can remove this part. 

Maybe adding a `CredentialState.Done` listener to log when the process completed will be helpful

**Review by TimoGlastra — CHANGES_REQUESTED** [review]
Great work @MosCD3!

I've left some comments. Do you think it would also be useful to handle creating a connection in this guide, or would you say that is a topic for another file?

**Comment by TimoGlastra** [comment]
Some practical notes:
- DCO is failing. I think it would be easiest to squash all commits into a single commit
- Formatting is failing. you can run `yarn format` to automatically fix this
- I'm not sure why there are so many changes that are unrelated to this PR. There's also some conflicts now. I'd try to merge with main to see if that fixes it

**Comment by MosCD3** [comment]
@TimoGlastra I would suggest adding all imports in one place, as adding each import to its respective blocks could confuse the developer into thinking that this code black is in a different file or component. 
Many developers tend to copy/past the code from the guide, so adding all imports in one step will make it easier

**Pull request closed without merging**